src/Flex.php

```diff
@@ -757,18 +757,31 @@ public function fetchRecipes(array $operations): array
             }
 
             $noRecipe = !isset($manifests[$name]) || (isset($manifests[$name]['not_installable']) && $manifests[$name]['not_installable']);
-            if ($noRecipe && 'symfony-bundle' === $package->getType()) {
-                $manifest = [];
-                $bundle = new SymfonyBundle($this->composer, $package, $job);
-                if (null === $devPackages) {
-                    $devPackages = array_column($this->composer->getLocker()->getLockData()['packages-dev'], 'name');
-                }
-                $envs = \in_array($name, $devPackages) ? ['dev', 'test'] : ['all'];
-                foreach ($bundle->getClassNames() as $class) {
-                    $manifest['manifest']['bundles'][$class] = $envs;
+            if ($noRecipe) {
+                $bundles = [];
+
+                $extra = $package->getExtra();
+                if (isset($extra['symfony']['bundles'])) {
+                    $extraBundles = (array) $extra['symfony']['bundles'];
```

Comment:
@nicolas-grekas should we check if bundles is array here instead of casting it?

Reply:
Maybe
```
$extraBundles = is_array($extra['symfony']['bundles']) ? $extra['symfony']['bundles'] : [];
```
?

Reply:
we could throw an exception if it's not an array
same in the foreach just after btw
and same if envs are not strings maybe also to be extra careful?

Reply:
I like throwing `InvalidArgumentException` exception. With envs it is simpler, as they could be a string. In that case `env` just will be converted to `[env]`. Which should be fine?

Reply:
better be strict if we validate something imho
we should validate the content of the array also

Reply:
Ok, so `InvalidArgumentException` in both cases?

Reply:
Or better https://www.php.net/manual/en/class.typeerror.php?

Reply:
type errors are for internal issues, aka badly written code
this is input validation

Reply:
Array checks have been added for extra bundles and their envs.